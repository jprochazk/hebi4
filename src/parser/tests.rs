use std::fs::read_to_string;

use crate::{parser::parse, token::tokenize};

globdir::globdir!("tests/inputs/*.hi", |path| {
    let input = read_to_string(path).unwrap();
    let tokens = tokenize(&input);
    let snapshot = match parse(&tokens) {
        Ok(ast) => format!("OK\n{ast:#?}"),
        Err(err) => format!("ERROR\n{}", err.render(&input)),
    };
    insta::assert_snapshot!(snapshot)
});
