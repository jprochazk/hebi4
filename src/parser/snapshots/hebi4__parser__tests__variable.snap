---
source: src/parser/tests.rs
expression: snapshot
---
OK
Root(
    [
        Var(
            Ident(
                "x",
            ),
            Int(
                0,
            ),
        ),
    ],
)
